File: BussinessObjects/Models/Payment.cs
namespace BussinessObjects.Models;

public class Payment
{
    public int PaymentId { get; set; }

    public int BookingId { get; set; }

    public int UserId { get; set; }

    public decimal Amount { get; set; }

    public string PaymentMethod { get; set; } = null!;

    public string? TransactionId { get; set; }

    public string PaymentStatus { get; set; } = null!;

    public DateTime? PaymentDate { get; set; }

    public string? PaymentGatewayResponse { get; set; }

    public virtual Booking Booking { get; set; } = null!;

    public virtual User User { get; set; } = null!;
    public enum PaymentStatusEnum
    {
        Pending,
        Success,
        Cancelled
    }
}

File: src/types/payment.ts
// Payment Types

export interface Payment {
  paymentId: number;
  bookingId: number;
  userId: number;
  helperId: number;
  amount: number;
  paymentStatus: PaymentStatus;
  paymentDate: string;
  transactionId: string | null;
  paymentMethod: string;
  bookingStatus: string;
  scheduledStartTime: string;
  scheduledEndTime: string;
  serviceName: string | null;
  helperName: string;
}

export type PaymentStatus = 
  | 'Pending'
  | 'Success'
  | 'Cancelled'
  | 'Failed';

export interface PaymentResponse {
  success: boolean;
  statusCode: number;
  message: string;
  data: Payment[];
  timestamp: string;
  requestId: string;
}

export interface PaymentInfo {
  paymentId: number;
  bookingId: number;
  userId: number;
  helperId: number;
  amount: number;
  paymentDate: string;
}

export interface PaymentInfoResponse {
  success: boolean;
  statusCode: number;
  message: string;
  data: PaymentInfo;
  timestamp: string;
  requestId: string;
} 